cpp/FileTransferAgent.cpp

```diff
@@ -283,19 +293,31 @@ void Snowflake::Client::FileTransferAgent::uploadFilesInParallel(std::string *co
       CXX_LOG_DEBUG("Sequential upload %d th file %s start", i, metadata->srcFileName.c_str());
       do
       {
-        RemoteStorageRequestOutcome outcome = uploadSingleFile(m_storageClient, metadata,
-          resultIndex);
+        RemoteStorageRequestOutcome outcome = RemoteStorageRequestOutcome::SUCCESS;
+        if(isPutFastFailEnabled() && !failedTransfers.empty())
+        {
+          m_executionResults->SetTransferOutCome(RemoteStorageRequestOutcome::SKIP_UPLOAD_FILE, resultIndex);
+          CXX_LOG_DEBUG("Sequential upload, put fast fail enabled, Skipping file.");
+          break;
+        }
+        outcome = uploadSingleFile(m_storageClient, metadata, resultIndex);
         if (outcome == RemoteStorageRequestOutcome::TOKEN_EXPIRED)
         {
+          CXX_LOG_DEBUG("Sequential upload %d th file %s renewToken", i, metadata->srcFileName.c_str());
           _mutex_lock(&m_parallelTokRenewMutex);
           this->renewToken(command);
           _mutex_unlock(&m_parallelTokRenewMutex);
         }
         else
         {
           if (outcome == RemoteStorageRequestOutcome::FAILED) {
-            CXX_LOG_DEBUG("Failed to upload %s ", metadata->srcFileName.c_str());
+            CXX_LOG_DEBUG("Sequential upload %s FAILED.", metadata->srcFileName.c_str());
             failedTransfers.append(metadata->srcFileName) + ", ";
+            //Fast fail, return when the first file fails to upload.
+            if(isPutFastFailEnabled())
+            {
+              CXX_LOG_DEBUG("Sequential upload, put fast fail enabled, Skip uploading rest of the files.");
+            }
```

Comment:
Shouldn't you have a break or return statement here?

Reply:
If returned from here then we do not throw the exception and ODBC driver will not know of the error. 
break statement in 325 already covers this. So did not add break again.